test/utils/general.spec.js

```diff
@@ -169,30 +169,30 @@ describe('services', () => {
       it('processes an array where the index field is a single value', () => {
         const inputArray = [
           { category: '1', value: 'a' },
-          { category: '1', value: 'b' }
+          { category: '1', value: 'b' },
         ];
-        expect(GeneralUtil.indexByField(inputArray, 'category')).toStrictEqual({ '1': inputArray });
+        expect(GeneralUtil.indexByField(inputArray, 'category')).toStrictEqual({ 1: inputArray });
       });
       it('processes an array where the index field does not exist', () => {
         const inputArray = [
           { category: '1', value: 'a' },
-          { category: '1', value: 'b' }
+          { category: '1', value: 'b' },
         ];
         expect(GeneralUtil.indexByField(inputArray, 'category2')).toStrictEqual({
-          undefined: inputArray
+          undefined: inputArray,
         });
       });
       it('processes an array where the index field can contain multiple values (array)', () => {
         const inputArray = [
           { category: ['1'], value: 'a' },
-          { category: ['1', '2'], value: 'b' }
+          { category: ['1', '2'], value: 'b' },
         ];
         expect(GeneralUtil.indexByField(inputArray, 'category')).toStrictEqual({
-          '1': [
+          1: [
             { category: ['1'], value: 'a' },
-            { category: ['1', '2'], value: 'b' }
+            { category: ['1', '2'], value: 'b' },
           ],
-          '2': [{ category: ['1', '2'], value: 'b' }]
+          2: [{ category: ['1', '2'], value: 'b' }],
```

Comment:
It should preserve the original type here, the function does not convert the type from string to int.

Reply:
I guess integers should work fine here. Both integers and strings are acceptable for JSON fields.
I've added an example codesnip here.
![Screenshot from 2024-04-03 13-09-17](https://github.com/StatTag/StatWrap/assets/96688318/9b1614d7-396e-4989-bbf6-00fc2cf462e7)

Reply:
Oh understood!

Reply:
But, I guess integers should work fine here. Both integers and strings are acceptable for JSON fields. And, the function is returning a JSON object.
Added an example codesnip here.
![Screenshot from 2024-04-03 13-09-17](https://github.com/StatTag/StatWrap/assets/96688318/f5fbdf67-07c6-432f-870a-e9ad51aaeb65)
Please correct me if am wrong though.

Reply:
You are correct that integers and strings are okay.  However, the function is taking the category values ('1' as a string) and returning that as part of the response.  So the function is not changing '1' to 1, and the test should reflect that we expect '1' returned as a string.